DataAccessLibrary/Repositories/Generic/Repository.cs

```diff
@@ -1,61 +1,64 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Data.SqlClient;
 using System.Linq;
-using System.Reflection;
 using System.Text;
 using System.Threading.Tasks;
-using Dapper;
-using DataAccessLibrary.Repositories.Generic;
-using Microsoft.Extensions.Configuration;
-using DataAccessLibrary;
+using DataAccessLibrary.Base;
 
 
 namespace DataAccessLibrary.Repositories.Generic
 {
     // TODO change T to eg. IDataTable - with eg. property TableName
     // so we can know more about the class
-    public class Repository<T> : IRepository<T> where T : class, new()
+    public class Repository<T> : IRepository<T>
+        where T : Domain
     {
         private readonly ISqlDataAccess _database;
+
         private readonly string _tableName;
+        
         private readonly List<string> _properties;
 
         public Repository(ISqlDataAccess database)
         {
-            _database = database;
+            _database = database ?? throw new ArgumentNullException(nameof(database));
             _tableName = typeof(T).GetField("TableName")?.GetValue(null)?.ToString();
             _properties = typeof(T).GetProperties().Select(p => p.Name).ToList();
         }
 
         // TODO add error catching (eg. what if type do not have table in repository)
-        public  Task<List<T>> GetAll()
+        public  Task<List<T>> GetAllAsync()
         {
             var query = $"Select * From {_tableName};";
+
             return _database.LoadData<T, object>(query, new {});
         }
 
-        public Task<T> GetById(Guid id)
+        public Task<T> GetByIdAsync(Guid id)
         {
             var query = $"Select * From {_tableName} Where [Id] = @Id;";
+
             return _database.LoadData<T, object>(query, new {Id = id })
                 .ContinueWith(t => t.Result.FirstOrDefault());
         }
-        public Task DeleteById(Guid id)
+        public Task DeleteByIdAsync(Guid id)
         {
             var query = $"Delete From {_tableName} Where [Id] = @Id;";
+
             return _database.SaveData(query, new { Id = id });
         }
 
-        public Task Insert(T entity)
+        public Task InsertAsync(T entity)
```

Comment:
Gdzie używasz tego `InsertAsync`? Czy przypadkiem `T` nie bierzesz z inputu usera?

Jeśli tak, to znowu masz problem z SQL Injection.

Wyobraź sobie, że w twoj body wysyłasz obiekt encji, for example:

```js
{
    "user":
    {
       "login": "ddd",
       "email": "aaaa@aaa.pl"
    }
}
```

Załóżmy, że na encjimasz flagę `IsAdmin`. 

Ty po inputach na swoim frontendzie możesz wyłączyć opcję ustawiania flagi `IsAdmin` przez usera. Jednak haker może strzelić sobie na twojego endpointa z np. Postmana. Doda sobie flagę `IsAdmin` na true i utworzy sobie sam konto admina. 

Jeśli `T entity` ustawiasz z inputu to bardzo mi przykro, ale powinnaś to zmienić na jakieś DTO. Nie mapuj inputu bezpośrednio na encje. Rób to sama i waliduj. Niestety będziesz chyba musiała przerobić te generyczną klase